### PR TITLE
[Coverity] Medium issues with GPU plugin

### PR DESCRIPTION
**Pull request by mhpanah**
### Details:
- Ignoring the following CIDs since the status is already fixed: 1548290, 1546891, 1546890, 1546888, 1546867, 1546862, 1546023, 1546007, 
- Ignoring the following CIDs since it is third party code: 1548253, 1545465
- Ignoring 1548233, believe it's false positive.
- Ignoring 1549003, per reviewers feedback it is false positive.  Updated Coverity status.
- This PR, fixes: 1548264

### Tickets:
 - [CVS-145083](https://jira.devtools.intel.com/browse/CVS-145083)

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/program_node.cpp [hunk 1]
```diff
@@ -1505,7 +1505,11 @@ void program_node::create_onednn_primitive_attributes(
                     oc_dim = static_cast<int>(desc.output_layout.get_partial_shape()[1].get_max_length());
                 else
                     oc_dim = static_cast<int>(desc.output_layout.get_tensor().feature.size());
-                post_ops.append_prelu(1 << std::max(0, oc_dim));
+                if (oc_dim >= 0) {
+                    post_ops.append_prelu(1 << oc_dim);
+                } else {
+                    throw std::runtime_error("oc_dim is negative, which is not allowed for bit shift operation.");
+                }
```
Comment (p-durandin) [inline src/plugins/intel_gpu/src/graph/program_node.cpp:1512]: Id better report as false positive, max from 0 and value can not be negative

Comment (mhpanah) [reply src/plugins/intel_gpu/src/graph/program_node.cpp:1512]: Updated Coverity status.

File: src/plugins/intel_gpu/src/graph/primitive_inst.cpp [hunk 2]
```diff
@@ -2028,7 +2028,8 @@ primitive_inst::primitive_inst(network & network, program_node const& node, bool
     , _inputs_memory_count(node.get_inputs_count())
     , _outputs_memory_count(node.get_outputs_count())
     , _fused_mem_count(node.get_fused_inputs_count())
-    , _fused_mem_offset((_fused_mem_count > 0 && node.get_first_fused_dep_idx() > 0) ? node.get_first_fused_dep_idx() : 0)
+    // Cast the result to uint64_t to avoid any potential underflow issues
+    , _fused_mem_offset((_fused_mem_count > 0 && node.get_first_fused_dep_idx() > 0) ? static_cast<uint64_t>(node.get_first_fused_dep_idx()) : 0)
```
[the remark this answers is not in the document]
Comment (mhpanah) [reply src/plugins/intel_gpu/src/graph/primitive_inst.cpp:2032]: Removed comment.

**Comment by p-durandin** [comment]
buid_jenkins

**Comment by p-durandin** [comment]
build_jenkins

File: src/plugins/intel_gpu/src/graph/program_node.cpp (hunk 1 above)
Comment (p-durandin) [reply src/plugins/intel_gpu/src/graph/program_node.cpp]: Can be resolved by cast to unsigned std::max(0, oc_dim)

**Review by p-durandin — APPROVED** [review]

File: src/plugins/intel_gpu/src/plugin/transformations/dynamic_quantize_fully_connected.cpp [hunk 3]
```diff
@@ -17,7 +17,8 @@
 namespace ov {
 namespace intel_gpu {
 
-DynamicQuantizeFullyConnected::DynamicQuantizeFullyConnected(uint64_t group_size) {
+DynamicQuantizeFullyConnected::DynamicQuantizeFullyConnected(uint64_t group_size)
+    : ov::pass::MatcherPass() {  // Explicitly call the base class constructor
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/plugin/transformations/dynamic_quantize_fully_connected.cpp:21]: Don't comment obvious things please

**Comment by p-durandin** [comment]
build_jenkins

**Comment by p-durandin** [comment]
build_jenkins

**Comment by p-durandin** [comment]
build_jenkins

**Review by vladimir-paramuzov — APPROVED** [review]

**Pull request merged**